### PR TITLE
[CIApp] - Fixes some agentless tags

### PR DESCRIPTION
**Pull request by tonyredondo**
This PR fixes some agentless tags and avoid sending unneeded ones.

### CODE REVIEW

**Comment by andrewlock** [comment]
## Benchmarks Report :snail:

Benchmarks for #2575 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf): 
* All benchmarks have **the same** speed
* All benchmarks have **the same** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`WriteAndFlushEnrichedTraces`|net472|871μs|3.2μs|12μs|0| 0|0|3.16&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|672μs|2.39μs|8.94μs|0| 0|0|2.57&nbsp;KB|
|#2575|`WriteAndFlushEnrichedTraces`|net472|901μs|5.04μs|34.2μs|0| 0|0|3.17&nbsp;KB|
|#2575|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|678μs|2.93μs|10.9μs|0| 0|0|2.57&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`SendRequest`|netcoreapp3.1|337μs|1.66μs|7.06μs|0.166| 0|0|19.74&nbsp;KB|
|#2575|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|#2575|`SendRequest`|netcoreapp3.1|336μs|1.28μs|4.95μs|0.171| 0|0|19.74&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`ExecuteNonQuery`|net472|1.83μs|8.6ns|32.2ns|0.0942| 0.000923|0|594&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`ExecuteNonQuery`|netcoreapp3.1|1.55μs|7.51ns|31.9ns|0.00853| 0|0|632&nbsp;B|
|#2575|`ExecuteNonQuery`|net472|1.82μs|8.98ns|41.2ns|0.0942| 0.00088|0|594&nbsp;B|
|#2575|`ExecuteNonQuery`|netcoreapp3.1|1.49μs|6.89ns|27.6ns|0.00865| 0|0|632&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`CallElasticsearch`|net472|2.86μs|16.4ns|122ns|0.125| 0|0|802&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`CallElasticsearch`|netcoreapp3.1|1.89μs|10.7ns|76.5ns|0.0114| 0|0|792&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`CallElasticsearchAsync`|net472|3.26μs|18.1ns|120ns|0.146| 0|0|939&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`CallElasticsearchAsync`|netcoreapp3.1|1.91μs|8.1ns|31.4ns|0.0134| 0|0|912&nbsp;B|
|#2575|`CallElasticsearch`|net472|2.92μs|15.2ns|94.8ns|0.125| 0|0|802&nbsp;B|
|#2575|`CallElasticsearch`|netcoreapp3.1|1.79μs|7.96ns|28.7ns|0.0107| 0|0|792&nbsp;B|
|#2575|`CallElasticsearchAsync`|net472|3.04μs|12.3ns|46.2ns|0.149| 0.00164|0|939&nbsp;B|
|#2575|`CallElasticsearchAsync`|netcoreapp3.1|1.83μs|9.06ns|39.5ns|0.0124| 0|0|912&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`ExecuteAsync`|net472|3.22μs|14.5ns|56.1ns|0.168| 0|0|1.06&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`ExecuteAsync`|netcoreapp3.1|1.99μs|5.42ns|20.3ns|0.0149| 0|0|1.03&nbsp;KB|
|#2575|`ExecuteAsync`|net472|3.46μs|20.1ns|186ns|0.166| 0|0|1.06&nbsp;KB|
|#2575|`ExecuteAsync`|netcoreapp3.1|1.94μs|5.04ns|17.4ns|0.0144| 0|0|1.03&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`SendAsync`|net472|7.42μs|47.5ns|470ns|0.355| 0|0|2.28&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`SendAsync`|netcoreapp3.1|4.83μs|23.3ns|90.2ns|0.0309| 0|0|2.21&nbsp;KB|
|#2575|`SendAsync`|net472|7.27μs|35.7ns|160ns|0.362| 0|0|2.28&nbsp;KB|
|#2575|`SendAsync`|netcoreapp3.1|4.89μs|13.1ns|48.9ns|0.0297| 0|0|2.21&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`EnrichedLog`|net472|3.49μs|18.2ns|87.4ns|0.228| 0|0|1.45&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`EnrichedLog`|netcoreapp3.1|3.21μs|13.7ns|47.4ns|0.0222| 0|0|1.53&nbsp;KB|
|#2575|`EnrichedLog`|net472|3.54μs|16.3ns|69.1ns|0.226| 0|0|1.45&nbsp;KB|
|#2575|`EnrichedLog`|netcoreapp3.1|3.23μs|14.4ns|53.8ns|0.0211| 0|0|1.53&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`EnrichedLog`|net472|303μs|1.28μs|4.96μs|0.454| 0.151|0|4.33&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`EnrichedLog`|netcoreapp3.1|248μs|1.43μs|11.5μs|0| 0|0|4.21&nbsp;KB|
|#2575|`EnrichedLog`|net472|302μs|921ns|3.32μs|0.45| 0.15|0|4.33&nbsp;KB|
|#2575|`EnrichedLog`|netcoreapp3.1|239μs|906ns|3.51μs|0| 0|0|4.21&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`EnrichedLog`|net472|7.72μs|35.5ns|138ns|0.505| 0|0|3.23&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`EnrichedLog`|netcoreapp3.1|6.56μs|17.5ns|63.1ns|0.049| 0|0|3.6&nbsp;KB|
|#2575|`EnrichedLog`|net472|8.27μs|43.1ns|211ns|0.504| 0|0|3.23&nbsp;KB|
|#2575|`EnrichedLog`|netcoreapp3.1|6.69μs|17.8ns|66.5ns|0.0496| 0|0|3.6&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`SendReceive`|net472|2.46μs|13.2ns|68.6ns|0.161| 0.00122|0|1.01&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`SendReceive`|netcoreapp3.1|2.13μs|9.28ns|34.7ns|0.0137| 0|0|1.01&nbsp;KB|
|#2575|`SendReceive`|net472|2.58μs|15.3ns|150ns|0.159| 0|0|1.01&nbsp;KB|
|#2575|`SendReceive`|netcoreapp3.1|2.15μs|8.77ns|34ns|0.0141| 0|0|1.01&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`EnrichedLog`|net472|6.62μs|29ns|116ns|0.291| 0|0|1.87&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`EnrichedLog`|netcoreapp3.1|5.41μs|14.6ns|54.7ns|0.0192| 0|0|1.49&nbsp;KB|
|#2575|`EnrichedLog`|net472|6.64μs|29.6ns|107ns|0.289| 0|0|1.87&nbsp;KB|
|#2575|`EnrichedLog`|netcoreapp3.1|5.37μs|20.9ns|81.1ns|0.0209| 0|0|1.49&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`StartFinishSpan`|net472|964ns|5.09ns|25.4ns|0.0717| 0|0|457&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`StartFinishSpan`|netcoreapp3.1|926ns|5.19ns|34.5ns|0.00631| 0|0|456&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`StartFinishScope`|net472|1.18μs|5.58ns|21.6ns|0.084| 0|0|538&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/67005d748c9c0acda77a359ffbfc7b5917aee3bf)|`StartFinishScope`|netcoreapp3.1|1.12μs|6.48ns|54.2ns|0.00754| 0|0|576&nbsp;B|
|#2575|`StartFinishSpan`|net472|930ns|4.89ns|23.4ns|0.0716| 0|0|457&nbsp;B|
|#2575|`StartFinishSpan`|netcoreapp3.1|908ns|4.62ns|20.2ns|0.00622| 0|0|456&nbsp;B|
|#2575|`StartFinishScope`|net472|1.22μs|6.85ns|46ns|0.0841| 0|0|538&nbsp;B|
|#2575|`StartFinishScope`|netcoreapp3.1|1.04μs|4.8ns|23.5ns|0.0079| 0|0|576&nbsp;B|
</details>

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2575 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/6c137b333c0b726deacf4d77b8f7302be2cc6938) will **not change** line coverage
:heavy_check_mark: Merging #2575 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/6c137b333c0b726deacf4d77b8f7302be2cc6938) will **not change** branch coverage
:no_entry: Merging #2575 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/6c137b333c0b726deacf4d77b8f7302be2cc6938) will will **increase** complexity by `8`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/6c137b333c0b726deacf4d77b8f7302be2cc6938) | #2575       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `12336` / `17227` | `12317` / `17227` |          |
| Lines %   | `72%`      | `71%`      |  `0%` :heavy_check_mark:  |
| Branches  | `6777` / `10141` | `6761` / `10149` |          |
| Branches %| `67%`      | `67%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `11349`      | `11357`        |  `8`  :no_entry:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=93445&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/93445/artifacts?artifactName=Code%20Coverage%20Report_93445&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2575](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=93560&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/93560/artifacts?artifactName=Code%20Coverage%20Report_93560&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/6c137b333c0b726deacf4d77b8f7302be2cc6938) | #2575       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `72%`      | `71%`       |  `0%` :heavy_check_mark:  |
| Branches %| `67%`      | `67%`       |  `0%` :heavy_check_mark:  |
| Complexity| `11349`      | `11357`       |  `8` :no_entry:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.Telemetry.JsonWebRequestTelemetryTransport](https://github.com/DataDog/dd-trace-dotnet/tree/c15e59d3f54348945d79de52b3356a5848da3dbc/tracer/src/Datadog.Trace/Telemetry/Transports/JsonWebRequestTelemetryTransport.cs) | `-28%` :no_entry: | `-57%` :no_entry:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=93445&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/93445/artifacts?artifactName=Code%20Coverage%20Report_93445&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2575](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=93560&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/93560/artifacts?artifactName=Code%20Coverage%20Report_93560&api-version=6.0&%24format=zip)

File: tracer/src/Datadog.Trace/Ci/Agent/MessagePack/CIEventMessagePackFormatter.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ internal class CIEventMessagePackFormatter : EventMessagePackFormatter<EventsPay
         private readonly byte[] _metadataBytes = StringEncoding.UTF8.GetBytes("metadata");
         private readonly byte[] _containerIdBytes = StringEncoding.UTF8.GetBytes("container_id");
         private readonly byte[]? _containerIdValueBytes;
-        private readonly byte[] _runtimeIdBytes = StringEncoding.UTF8.GetBytes("runtime_id");
+        private readonly byte[] _runtimeIdBytes = StringEncoding.UTF8.GetBytes(Trace.Tags.RuntimeId);
```
Comment (zacharycmontoya) [inline tracer/src/Datadog.Trace/Ci/Agent/MessagePack/CIEventMessagePackFormatter.cs:22]: This changes the tag from `"runtime_id"` to `"runtime-id"`, will that cause any issues?

Comment (tonyredondo) [reply tracer/src/Datadog.Trace/Ci/Agent/MessagePack/CIEventMessagePackFormatter.cs:22]: It’s part of the fix. I was sending a bad key.

**Review by zacharycmontoya — APPROVED** [review]

**Review by kevingosse — APPROVED** [review]

**Pull request merged**